src/App.js

```diff
@@ -1,127 +1,102 @@
-              <div class="col button-single">
-                2
-              </div>
-              <div class="col button-single">
-                3
-              </div>
-              <div class="col button-single" id="equals">
-                =
-              </div>
-            </div> <!-- end of fourth-row -->
-            <div class="row fifth-row">
-              <div class="col button-single">
+            <div className='display'>
+              <div className='mem-input'>
                 0
               </div>
-              <div class="col button-single">
+              <div className='act-input'>
                 0
               </div>
-              <div class="col button-single">
-                .
-              </div>
-              <div class="col button-single" id="equals">
-                =
+            </div>
+            <div className='buttons-all'>
+              <div className='container'>
+                <div className='row first-row'>
+                  <div className='col button-single button-special' id='AC'>
+                    AC
+                  </div>
+                  <div className='col button-single button-special' id='CE'>
+                    CE
+                  </div>
+                  <div className='col button-single' id='divide'>
+                    ÷
+                  </div>
+                  <div className='col button-single' id='multiply'>
+                    x
+                  </div>
+                </div>
+                <div className='row second-row'>
+                  <div className='col button-single'>
+                    7
+                  </div>
+                  <div className='col button-single'>
+                    8
+                  </div>
+                  <div className='col button-single'>
+                    9
+                  </div>
+                  <div className='col button-single' id='minus'>
+                    -
+                  </div>
+                </div>
+                <div className='row third-row'>
+                  <div className='col button-single'>
+                    4
+                  </div>
+                  <div className='col button-single'>
+                    5
+                  </div>
+                  <div className='col button-single'>
+                    6
+                  </div>
+                  <div className='col button-single' id='plus'>
+                    +
+                  </div>
+                </div>
+                <div className='row fourth-row'>
+                  <div className='col button-single'>
+                    1
+                  </div>
+                  <div className='col button-single'>
+                    2
+                  </div>
+                  <div className='col button-single'>
+                    3
+                  </div>
+                  <div className='col button-single' />
+                  <div className='button-single' id='equals'>
```

Comment:
I took the `=` out of the document flow and added the dummy `<div className='col button-single' />` above (with no content) to make sure that flexbox flexed the right width. Otherwise, the widths would never be fully accurate (you'll get accurate widths if you're always making columns from the same row width - this changes if you don't have a dummy element). This is actually a super tricky problem to solve with html and css.